spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/PulsarAutoConfigurationTests.java

```diff
@@ -154,6 +157,16 @@ void customPulsarListenerAnnotationBeanPostProcessorIsRespected() {
 						.isSameAs(listenerAnnotationBeanPostProcessor));
 	}
 
+	@Test
+	void customProducerInterceptorIsUsedInPulsarTemplate() {
```

Comment:
This looks great @alpreu . It may be helpful to add another variant that configures in 2 interceptors for sanity check. There is a variant of `withBean` that takes a `BeanDefinitionCustomizer` but that does not let you set the `@Order` for the bean. However, you can still use the `ApplicationContextRunner` approach w/ a slight twist like:
```java
	@Test
	void someTest() {
		this.contextRunner
				.withUserConfiguration(InterceptorTestConfiguration.class)
				.run((context) -> assertThat(context).hasNotFailed().hasBean("interceptorOne"));
	}

	@Configuration(proxyBeanMethods = false)
	static class InterceptorTestConfiguration {

		@Bean
		@Order(100)
		String interceptorOne() {
			return ...
		}
                 ....
	}
```
You may already know about this though. 

Also, this is not critical and if you don't have time to get to it I would be happy to add it during merge etc.. Just lmk.

Reply:
How would you retrieve the bean instances from the configuration to confirm the order in the list? Or would you just check that there are two interceptors set in the Template? If you can add this part I'd be interested in taking a look at the solution :)

Reply:
I added the test [here](https://github.com/spring-projects-experimental/spring-pulsar/commit/664c62eb3672ea08c080f2a636537dcb8c224e89#diff-89ca4830da9a2117dd3d4c118fffa9f7c67e601248e9aa7c5a231568855d998eR226). I used a static instance var to identify the interceptors. I know, accessing the statics like this does feel gross. On the flip side, the alternative that I know of is to make override some identifying method on the producer (see below for `toString`) also feels gross. Of the two choices I went w/ the former mostly because it is less code and a bit easier to follow than some magic toString override. 
```java


@Test
void customProducerInterceptorsOrderedProperly() {
  this.contextRunner
      .withUserConfiguration(InterceptorTestConfiguration.class)
      .run((context -> assertThat(context).hasNotFailed().getBean(PulsarTemplate.class)
          .extracting("interceptors")
          .asInstanceOf(InstanceOfAssertFactories.list(ProducerInterceptor.class))
          .extracting(ProducerInterceptor::toString)
          .containsExactly("ProducerInterceptor(Bar)", "ProducerInterceptor(Foo)")));
}

@Configuration(proxyBeanMethods = false)
static class InterceptorTestConfiguration {
  @Bean
  @Order(200)
  ProducerInterceptor interceptorFoo() {
    ProducerInterceptor interceptor = mock(ProducerInterceptor.class);
    when(interceptor.toString()).thenReturn("ProducerInterceptor(Foo)");
    return interceptor;
  }

  @Bean
  @Order(100)
  ProducerInterceptor interceptorBar() {
    ProducerInterceptor interceptor = mock(ProducerInterceptor.class);
    when(interceptor.toString()).thenReturn("ProducerInterceptor(Bar)");
    return interceptor;
  }
}
```

Reply:
Ah I see. Yeah, the toString-way is a bit hacky I agree